runelite-client/src/main/java/net/runelite/client/plugins/theatre/rooms/nylocas/NyloHandler.java

```diff
@@ -368,6 +393,127 @@ public void onGameTick()
 		}
 	}
 
+	private int checkAttackStyle(String itemName)
+	{
+		switch (itemName.toLowerCase())
+		{
+			case "toxic blowpipe":
+			case "twisted bow":
+			case "craw's bow":
+				return 1;
+			case "abyssal whip":
+			case "abyssal tentacle":
+			case "scythe of vitur":
+			case "scythe of vitur (uncharged)":
+			case "ham joint":
+			case "bandos godsword":
+			case "bandos godsword (or)":
+			case "dragon warhammer":
+			case "dragon claws":
+			case "event rpg":
+			case "ghrazi rapier":
+			case "blade of saeldor":
+			case "crystal halberd":
+			case "dragon scimitar":
+				return 2;
+			case "kodai wand":
+			case "master wand":
+			case "trident of the seas":
+			case "trident of the swamp":
+			case "sanguinesti staff":
+			case "iban's staff":
+			case "iban's staff (u)":
+			case "trident of the swamp (e)":
+			case "trident of the seas (e)":
+				return 3;
+			case "red chinchompa":
+			case "chinchompa":
+			case "black chinchompa":
+			case "armadyl crossbow":
+			case "dragon crossbow":
+			case "rune crossbow":
+				return 4;
+			case "avernic defender":
+			case "dragon defender":
+			case "dragon defender (t)":
+				if (attackStyle == 1)
+				{
+					return 2;
+				}
+				else
+				{
+					return attackStyle;
+				}
+			default:
+				return attackStyle;
+		}
+	}
+
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (!plugin.isNylocasMenuSwap())
+		{
+			attackStyle = 0;
+			removeMenuSwaps();
+			return;
+		}
+		final String menuOption = event.getOption();
+		if (!menuOption.equalsIgnoreCase("equip") && !menuOption.equalsIgnoreCase("attack"))
+		{
+			return;
+		}
+
+		if (attackStyle == 0)
+		{
+			String itemName;
+			if (client.getLocalPlayer() == null
+				|| client.getLocalPlayer().getPlayerAppearance() == null
+				|| client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN) != null
+				|| client.getViewportWidget() == null
+				|| client.getLocalPlayer().getPlayerAppearance() == null
+				|| client.getLocalPlayer().getPlayerAppearance().getEquipmentId(KitType.WEAPON) == 0
+			)
+			{
+				return;
+			}
+			final int weapon = ObjectUtils.defaultIfNull((client.getLocalPlayer().getPlayerAppearance().getEquipmentId(KitType.WEAPON)), 0);
+
+			if (weapon == 0) {
+				return;
+			}
+			ItemDefinition equippedWeapon = itemManager.getItemDefinition(weapon);
+			itemName = equippedWeapon.getName();
+			if (itemName != null) {
+				attackStyle = checkAttackStyle(itemName);
+			}
+		}
+		if (event.getOption().equalsIgnoreCase("attack"))
```

Comment:
whats the use of attack being getting up to here?

Reply:
to set a menu entry swap if no equipment has changed yet, this allows you to menu entry swap without having to wait the tick it takes to equip items.

Reply:
changed it so every option will make it to swaps if its null